src/main/java/com/epam/prejap/teatrees/game/Playfield.java

```diff
@@ -16,33 +13,35 @@
     private int col;
 
     public Playfield(int rows, int cols, BlockFeed feed, Printer printer) {
-        this.rows = rows;
-        this.cols = cols;
         this.feed = feed;
         this.printer = printer;
-        grid = new byte[this.rows][this.cols];
+        grid = new Grid(rows, cols);
+    }
+    Playfield(Grid grid, BlockFeed feed, Printer printer) {
+        this.feed = feed;
+        this.printer = printer;
+        this.grid = grid;
     }
 
     public void nextBlock() {
         block = feed.nextBlock();
         row = 0;
-        col = (cols - block.cols()) / 2;
-        show();
+        col = (grid.getNumberOfCols() - block.cols()) / 2;
```

Comment:
The getter could be replaced with a direct access to a package-private field, since Grid is actually a part of Playfield (composition).